src/core/obj.js

```diff
@@ -135,6 +135,19 @@ var Dict = (function DictClosure() {
       return Promise.resolve(value);
     },
 
+    // Same as get(), but dereferences all elements if the result is an Array.
+    getArray: function Dict_getArray(key1, key2, key3) {
+      var value = this.get(key1, key2, key3);
+      if (!isArray(value)) {
+        return value;
+      }
+      var valueLen = value.length, valueArray = new Array(valueLen);
```

Comment:
At this point we know that `value` is an array, so wouldn't it suffice just to override its elements in the `for` loop below instead of creating a new array? That we we avoid array allocations by reusing the `value` array, which can be done because `this.xref.fetchIfRef()` maintains the original element if it is *not* a reference.